plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/validation/SCTValidationJob.java

```diff
@@ -40,20 +34,18 @@
 import org.yakindu.sct.ui.editor.DiagramActivator;
 
 import com.google.inject.Inject;
-import com.google.inject.Singleton;
 
 /**
  * 
  * @author andreas muelder - Initial contribution and API
  * 
  */
-@Singleton
-public class SCTValidationJob extends Job implements IMarkerType {
+public class SCTValidationJob extends Job {
 
 	@Inject
 	private IResourceValidator validator;

```

Comment:
By running the validation with the IResourceValidator you get the overhead of issue creation and this results in your API design for IValidationIssueProcessor.
I suggest not to work with issues at all. Just use the Diagnositcian for validation and work completely with Diagnostic.
You will get FeatureBasedDiagnositcs which will be more easily be connectable to the target EObjects.

But this is only a personal preference of mine.